aten/src/ATen/TensorIndexing.h

```diff
@@ -39,50 +39,65 @@ struct TORCH_API Slice final {
  public:
   // This mirrors `__PySlice_Unpack` in torch/csrc/utils/python_compat.h
   Slice(
-      c10::optional<int64_t> start_index = c10::nullopt,
-      c10::optional<int64_t> stop_index = c10::nullopt,
-      c10::optional<int64_t> step_index = c10::nullopt) {
+      c10::optional<c10::SymInt> start_index = c10::nullopt,
+      c10::optional<c10::SymInt> stop_index = c10::nullopt,
+      c10::optional<c10::SymInt> step_index = c10::nullopt) {
     if (!step_index.has_value()) {
-      step_ = 1;
+      step_ = c10::SymInt(1);
     } else {
       step_ = step_index.value();
-      TORCH_CHECK_VALUE(step_ != 0, "slice step cannot be zero");
-
-      // Here step might be -INDEX_MAX-1; in this case we replace it
-      // with -INDEX_MAX.  This doesn't affect the semantics, and it
-      // guards against later undefined behaviour resulting from code that
-      // does "step = -step" as part of a slice reversal.
-      if (step_ < -INDEX_MAX)
-        step_ = -INDEX_MAX;
+      if (!step_.is_symbolic()) {
+        TORCH_CHECK_VALUE(step_.expect_int() != 0, "slice step cannot be zero");
+        // Here step might be -INDEX_MAX-1; in this case we replace it
+        // with -INDEX_MAX.  This doesn't affect the semantics, and it
+        // guards against later undefined behaviour resulting from code that
+        // does "step = -step" as part of a slice reversal.
+        if (step_.expect_int() < -INDEX_MAX) {
+          step_ = c10::SymInt(-INDEX_MAX);
+        }
+      }
     }
     if (!start_index.has_value()) {
-      start_ = step_ < 0 ? INDEX_MAX : 0;
+      TORCH_CHECK_VALUE(!step_.is_symbolic(), "slice step cannot be symbolic");
```

Comment:
If you expect this here, then you don't need to handle the case above?

Reply:
updated